src/core/qgstemporalrangeobject.cpp

```diff
@@ -0,0 +1,63 @@
+/***************************************************************************
+                         qgstemporalrangeobject.cpp
+                         ---------------
+    begin                : January 2020
+    copyright            : (C) 2020 by Samweli Mwakisambwe
+    email                : samweli at kartoza dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+
+#include "qgstemporalrangeobject.h"
+#include "qgsrendercontext.h"
+
+QgsTemporalRangeObject::QgsTemporalRangeObject()
+{
+}
+
+QgsTemporalRangeObject::QgsTemporalRangeObject( bool enabled )
+  : mTemporal( enabled )
+{
+}
+
+QgsTemporalRangeObject::QgsTemporalRangeObject( const QgsRenderContext &rh )
+  : mTemporal( rh.isTemporal() )
+  , mDateTimeRange( rh.temporalRange() )
+{
+}
+
+void QgsTemporalRangeObject::setIsTemporal( bool enabled )
+{
+  mTemporal = enabled;
+}
+
+bool QgsTemporalRangeObject::isTemporal() const
+{
+  return mTemporal;
+}
+
+void QgsTemporalRangeObject::setTemporalRange( const QgsDateTimeRange &dateTimeRange )
+{
+  if ( dateTimeRange == mDateTimeRange )
```

Comment:
I'd drop this condition -- it could lead to strange behavior, because the `setIsTemporal( true );` below won't be enforced.